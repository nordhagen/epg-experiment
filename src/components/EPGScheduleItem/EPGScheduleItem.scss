.EPGScheduleItem {
  display: none;
  flex-direction: column;
  justify-content: center;
  // border: 1px solid red;
  flex-grow: 1;
  height: 100%;
  background: black;
  position: relative;
  z-index: 2;

  &:hover {
    background: #333;
  }

  h3 {
    margin: 0;
    order: 2;
  }
}
.EPGScheduleItemLive {
  display: flex;
}

.timeSlot {
  font-size: 12px;
  color: #666;
  margin: 0;
}

.progressBar {
  order: 3;
  background: #666;
  height: 5px;
  margin-top: 10px;

  .track {
    background: #EAB44A;
    height: 100%;
    transition: width 250ms ease-in-out;
  }
}