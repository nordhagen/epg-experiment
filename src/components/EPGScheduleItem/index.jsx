import React from 'react'
import moment from 'moment'
import styles from './EPGScheduleItem'

const FMT = 'HH:MM'

const EPGScheduleItem = props => {
  let now = moment(props.time)
  let startTime = moment(props.start)
  let endTime = moment(props.end)
  let isLive = now.isBetween(startTime, endTime)
  let className = styles.EPGScheduleItem
  let trackStyle

  if (isLive) {
    className += ' ' + styles.EPGScheduleItemLive
    let duration = endTime.diff(startTime)
    let elapsed = endTime.diff(now)
    let progress = 1 - elapsed / duration
    trackStyle = {
      width: Math.round(progress * 100) + '%'
    }
  }

  return (
    <li
      className={className}
      onClick={e => {
        props.selectDelegate(props)
      }}
    >
      <h3>{props.title}</h3>
      <p className={styles.timeSlot}>
        {moment(props.start).format(FMT)} â€“ {moment(props.end).format(FMT)}
      </p>

      {isLive && (
        <div className={styles.progressBar}>
          <div style={trackStyle} className={styles.track} />
        </div>
      )}
    </li>
  )
}

export default EPGScheduleItem
