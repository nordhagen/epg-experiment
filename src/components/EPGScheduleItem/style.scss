@import '../../global.scss';

.EPGScheduleItem {
  display: inline-flex;
  vertical-align: top;
  flex-direction: column;
  justify-content: center;
  height: 100%;
  background: black;
  position: relative;
  z-index: 2;
  cursor: pointer;

  @include media(mobile) {
    display: none;
    width: 100% !important;
    &:hover {
      background: #333;
    }
  }
  h3 {
    margin: 0 10px;
    order: 2;
    @include media(phablet) {
      order: 1;
    }
  }

  @include media(phablet) {
    border-bottom: 1px solid #333;
    border-left: 1px solid #333;

    &:after {
      content: '';
      display: block;
      position: absolute;
      top: 0;
      right: 0;
      width: 30px;
      height: 100%;
      @include gradient(left, rgba(0, 0, 0, 0), rgba(0, 0, 0, 255));
    }
  }
}
.EPGScheduleItemLive {
  display: flex;
  @include media(phablet) {
    background: #333;
  }
}

.timeSlot {
  font-size: 12px;
  color: #666;
  margin: 0 10px;
  @include media(phablet) {
    order: 2;
  }
}

.progressBar {
  order: 3;
  background: #666;
  height: 5px;
  margin: 10px 10px 0 10px;

  .track {
    background: #eab44a;
    height: 100%;
    transition: width 250ms ease-in-out;
  }
}
