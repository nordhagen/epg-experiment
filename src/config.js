// The mock data has EPG entries for all 24 hours of 2018-10-26.
// Starting the virtual clock at 18:00 in the evening works well.
export const STARTING_TIME_OF_DAY = '2018-10-26T18:00:00+02:00'

// How often (in ms) to update the global clock.
export const CLOCK_TICK_FREQUENCY = 1000

// How fast time flies in relation to realtime.
// Elapsed clock time will multiply by this value as
// often as CLOCK_TICK_FREQUENCY
export const TIME_MULTIPLIER = 1

// Set to true to display the current clock time in the interface.
export const SHOW_CLOCK = false