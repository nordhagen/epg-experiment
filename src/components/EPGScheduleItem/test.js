import React from 'react'
import { STARTING_TIME_OF_DAY } from '../../config'
import EPGScheduleItem from '../EPGScheduleItem'
import renderer from 'react-test-renderer'

const data = {
  title: 'Vikings',
  id: 'dummy_program_id',
  start: '2018-10-26T12:50:00+02:00',
  end: '2018-10-26T13:30:00+02:00'
}

test('It renders correctly', () => {
  const c = renderer.create(
    <EPGScheduleItem
      key={data.id}
      time={STARTING_TIME_OF_DAY}
      selectDelegate={() => {}}
      {...data}
    />
  )

  let tree = c.toJSON()
  expect(tree).toMatchSnapshot()
})
