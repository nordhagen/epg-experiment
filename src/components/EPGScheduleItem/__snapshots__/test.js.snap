// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders live item with progress bar 1`] = `
<li
  className="undefined undefined"
  onClick={[Function]}
  style={
    Object {
      "width": "53.33333333333333%",
    }
  }
>
  <h3>
    Vikings
  </h3>
  <p>
    12:10
     – 
    13:10
  </p>
  <div>
    <div
      style={
        Object {
          "width": "25%",
        }
      }
    />
  </div>
</li>
`;

exports[`It renders past item with no progress bar 1`] = `
<li
  onClick={[Function]}
  style={
    Object {
      "width": "53.33333333333333%",
    }
  }
>
  <h3>
    Vikings
  </h3>
  <p>
    12:10
     – 
    13:10
  </p>
</li>
`;
